src/CSharp/CodeCracker/Refactoring/IntroduceFieldFromConstructorCodeFixProvider.cs

```diff
@@ -0,0 +1,63 @@
+        {
+            var root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+            var diagnostic = context.Diagnostics.First();
+            var diagnosticSpan = diagnostic.Location.SourceSpan;
+            var parameterName = diagnostic.GetMessage().Split(':')[1].Trim();
+            var declaration = root.FindToken(diagnosticSpan.Start).Parent.AncestorsAndSelf().OfType<ConstructorDeclarationSyntax>().First();
+            context.RegisterFix(CodeAction.Create($"Introduce field: {parameterName} from constructor.", c => IntroduceFieldFromConstructorAsync(context.Document, declaration, parameterName, c)), diagnostic);
+        }
+
+        private async Task<Document> IntroduceFieldFromConstructorAsync(Document document, ConstructorDeclarationSyntax constructorStatement, string parameterName, CancellationToken cancellationToken)
+        {
+            var root = await document.GetSyntaxRootAsync(cancellationToken);
+
+            var oldClass = constructorStatement.FirstAncestorOrSelf<ClassDeclarationSyntax>();
+            var fieldMembers = oldClass.Members.OfType<FieldDeclarationSyntax>();
+            var fieldName = parameterName;
+
+            if (fieldMembers.Any(p => p.Declaration.Variables.First().Identifier.Text == fieldName))
+            {
+                fieldName += "1";
+            }
+
+            var assignmentField = SyntaxFactory.ExpressionStatement(SyntaxFactory.AssignmentExpression(SyntaxKind.SimpleAssignmentExpression,
+                                               SyntaxFactory.MemberAccessExpression(SyntaxKind.SimpleMemberAccessExpression, SyntaxFactory.ThisExpression(),
+                                               SyntaxFactory.IdentifierName(fieldName)), SyntaxFactory.IdentifierName(parameterName)));
+
+            var newField = SyntaxFactory.FieldDeclaration(SyntaxFactory.VariableDeclaration(SyntaxFactory.PredefinedType(SyntaxFactory.Token(SyntaxKind.IntKeyword)))
+                              .WithVariables(SyntaxFactory.SingletonSeparatedList<VariableDeclaratorSyntax>(SyntaxFactory.VariableDeclarator(SyntaxFactory.Identifier(fieldName)))))
+                              .WithModifiers(SyntaxFactory.TokenList(new[] { SyntaxFactory.Token(SyntaxKind.PrivateKeyword), SyntaxFactory.Token(SyntaxKind.ReadOnlyKeyword) }))
+                              .WithAdditionalAnnotations(Formatter.Annotation);
+
+            var newConstructor = constructorStatement.WithBody(constructorStatement.Body.AddStatements(assignmentField));
+            var newClassConstructor = oldClass.ReplaceNode(constructorStatement, newConstructor);
+            var newClass = newClassConstructor.WithMembers(newClassConstructor.Members.Insert(0, newField))
+                           .WithoutAnnotations(Formatter.Annotation);
+            var newRoot = root.ReplaceNode(oldClass, newClass);
+            return document.WithSyntaxRoot(newRoot);
+        }
+    }
+}
```

Comment:
I guess this will not work with Fix all. Can you test that and make sure? Check the